packages/camlrack/camlrack.0.9.0/opam

```diff
@@ -0,0 +1,43 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "S-Expression parsing for OCaml"
+description: """
+'Camlrack' provides the ability to parse strings into
+S-expressions and vice versa. It is intended to be simple and minimal, yet
+fully-featured."""
+maintainer: ["Pierce Darragh <pierce.darragh@gmail.com>"]
+authors: ["Pierce Darragh <pierce.darragh@gmail.com>"]
+license: "MIT"
+tags: ["S-Expressions" "parsing"]
+homepage: "https://github.com/pdarragh/camlrack"
+doc: "https://github.com/pdarragh/camlrack"
+bug-reports: "https://github.com/pdarragh/camlrack/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ocaml" {>= "4.12"}
```

Comment:
```suggestion
  "ocaml" {>= "4.12"}
  "ppx_camlrack" {= version & with-test & post}
```

Reply:
otherwise you need to move all the tests on the ppx package

Reply:
@kit-ty-kate it fails still with post, do we have a better solution?

Reply:
`post` is not the solution here. It will just make the dependency install after this package (which is going to make it fail anyway).

The solution here is to modify the dune files in the source and add the correct `(package ...)` field to each `(test ...)`. By default all of them will be run regardless of the `-p` given.

Reply:
So if I add the `(package ...)` to the tests, should I also do `"ppx_camlrack" {= version & with-test & post}`, or is that no longer necessary? (Sorry, I've never distributed a package on opam before.)

Reply:
It depends. If it is `(package ppx_camlrack)` no, otherwise yes